File: mobile/src/types/offender.ts
export type OffenderSearchRequest = {
  last_name?: string | null;
  first_name_initial?: string | null;
  tdcj_number?: string | null;
  sid?: string | null;
  page?: number;
};

export type OffenderSummary = {
  sid: string;
  name: string;
  tdcj_number: string | null;
  race: string | null;
  gender: string | null;
  projected_release_date: string | null;
  unit: string | null;
  age: number | null;
  detail_url: string;
};

export type OffenderSearchPagination = {
  current_page: number;
  total_pages: number;
  has_more: boolean;
};

export type OffenderSearchResponse = {
  results: OffenderSummary[];
  pagination: OffenderSearchPagination;
  source: string;
  retrieved_at: string;
};

export type OffenseHistoryEntry = {
  offense_date: string | null;
  offense: string | null;
  sentence_date: string | null;
  county: string | null;
  case_number: string | null;
  sentence_length: string | null;
};

export type OffenderDetail = {
  sid: string;
  tdcj_number: string | null;
  name: string | null;
  race: string | null;
  gender: string | null;
  age: number | null;
  maximum_sentence_date: string | null;
  current_facility: string | null;
  projected_release_date: string | null;
  parole_eligibility_date: string | null;
  visitation_eligible: boolean | null;
  visitation_eligible_raw: string | null;
  scheduled_release_date_text: string | null;
  scheduled_release_type_text: string | null;
  scheduled_release_location_text: string | null;
  parole_review_url: string | null;
  offense_history: OffenseHistoryEntry[];
  source: string;
  retrieved_at: string;
  source_note: string;
};

export type ParoleBoardOffice = {
  office_code: string;
  office_name: string;
  address_lines: string[];
  city: string;
  state: string;
  postal_code: string;
  phone: string | null;
  notes: string | null;
};

File: backend/app/schemas/offender.py
from __future__ import annotations

from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class OffenderSearchRequest(BaseModel):
    last_name: Optional[str] = None
    first_name_initial: Optional[str] = None
    tdcj_number: Optional[str] = None
    sid: Optional[str] = None
    page: int = 1


class OffenderSearchResult(BaseModel):
    name: str
    sid: str
    tdcj_number: Optional[str] = None
    race: Optional[str] = None
    gender: Optional[str] = None
    projected_release_date: Optional[str] = None
    unit: Optional[str] = None
    age: Optional[int] = None
    detail_url: str


class OffenderSearchPagination(BaseModel):
    current_page: int
    total_pages: int
    has_more: bool


class OffenderSearchResponse(BaseModel):
    results: list[OffenderSearchResult]
    pagination: OffenderSearchPagination
    source: str
    retrieved_at: datetime


class OffenseHistoryEntry(BaseModel):
    offense_date: Optional[str] = None
    offense: Optional[str] = None
    sentence_date: Optional[str] = None
    county: Optional[str] = None
    case_number: Optional[str] = None
    sentence_length: Optional[str] = None


class OffenderDetailResponse(BaseModel):
    sid: str
    tdcj_number: Optional[str] = None
    name: Optional[str] = None
    race: Optional[str] = None
    gender: Optional[str] = None
    age: Optional[int] = None
    maximum_sentence_date: Optional[str] = None
    current_facility: Optional[str] = None
    projected_release_date: Optional[str] = None
    parole_eligibility_date: Optional[str] = None
    visitation_eligible: Optional[bool] = None
    visitation_eligible_raw: Optional[str] = None
    scheduled_release_date_text: Optional[str] = None
    scheduled_release_type_text: Optional[str] = None
    scheduled_release_location_text: Optional[str] = None
    parole_review_url: Optional[str] = None
    offense_history: list[OffenseHistoryEntry]
    source: str
    retrieved_at: datetime
    source_note: str
